From 63ae7ebb0c6177efed46734061fb59c458a47a26 Mon Sep 17 00:00:00 2001
From: Saul Wold <sgw@linux.intel.com>
Date: Tue, 19 Jan 2016 09:49:19 -0800
Subject: [PATCH 2/2] configure.ac/Makefile.am: Check for input.h and
 input-event-codes.h

Add INPUT_HEADER to know which header file to use because it changed
in the 4.4 kernel code.

Set INPUT_HEADER with default linux/input.h if it exists and if the
newer input-events-codes.h exists from 4.4 use it instead.

Upstream-Status: Inappropriate [Upstream unsupported]

Signed-off-by: Saul Wold <sgw@linux.intel.com>
---
 Makefile.am  | 2 +-
 configure.ac | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 2e32e69..04eb194 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -591,7 +591,7 @@ dist_udevkeymapforcerel_DATA = \
 	src/keymap/force-release-maps/samsung-other \
 	src/keymap/force-release-maps/common-volume-keys
 
-src/keymap/keys.txt: $(INCLUDE_PREFIX)/linux/input.h
+src/keymap/keys.txt: $(INCLUDE_PREFIX)/$(INPUT_HEADER)
 	$(AM_V_at)mkdir -p src/keymap
 	$(AM_V_GEN)$(AWK) '/^#define.*KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' < $< | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
 
diff --git a/configure.ac b/configure.ac
index 0500313..9674620 100644
--- a/configure.ac
+++ b/configure.ac
@@ -154,8 +154,10 @@ AS_IF([test "x$enable_keymap" = "xyes"], [
        if test -z "$GPERF"; then
               AC_MSG_ERROR([gperf is needed])
        fi
+       
+       AC_CHECK_HEADER([linux/input.h], [AC_SUBST([INPUT_HEADER], [$(echo 'linux/input.h')])], AC_MSG_ERROR([kernel headers not found]))
+       AC_CHECK_HEADER([linux/input-event-codes.h], [AC_SUBST([INPUT_HEADER], [$(echo 'linux/input-event-codes.h')])], [:])
 
-       AC_CHECK_HEADER([linux/input.h], [:], AC_MSG_ERROR([kernel headers not found]))
        AC_SUBST([INCLUDE_PREFIX], [$(echo '#include <linux/input.h>' | eval $ac_cpp -E - | sed -n '/linux\/input.h/ {s:.*"\(.*\)/linux/input.h".*:\1:; p; q}')])
 ])
 AM_CONDITIONAL([ENABLE_KEYMAP], [test "x$enable_keymap" = "xyes"])
-- 
2.5.0

